liana/method/_pipe_utils/_get_mean_perms.py

```diff
@@ -49,51 +50,81 @@ def _get_means_perms(adata: anndata.AnnData,
     for ct_idx, label in enumerate(labels):
         labels_mask[:, ct_idx] = adata.obs.label == label
 
-    # Perm should be a cube /w dims: n_perms x idents x n_genes
-    perms = _generate_perms_cube(adata.X, n_perms, labels_mask, seed, agg_fun, verbose)
+    # generate two cube for metabolites and genes
+    if met:
+        perms = _generate_perms_cube(adata.X, n_perms, labels_mask, seed, agg_fun, verbose, met=met, Y=adata.obsm['metabolite_abundance'])
+        perms_ligand = perms[0]
+        perms_receptor = perms[1]
 
-    return perms
+        return perms_ligand, perms_receptor
```

Comment:
Wouldn't it be cleaner to just call the function twice - i.e. move the flag a scope above, I prefer to have the function have a singular purpose

Reply:
good point, lets discuss in person